### PR TITLE
256 convert care home extract tests script

### PR DESCRIPTION
**Pull request by lizihao-anu**
linked with #256 
Hope it makes sense. I know there might be quite a few things to improve. So feel free to comment.

### CODE REVIEW

**Review by Moohan — COMMENTED** [review]
Looks good, happy to approve once that build warning is resolved 🙂

File: R/produce_slf_ch_tests.R [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#' Care Home All Episodes Tests
+#'
+#' @description Produce the test for the Care Home
+#'
+#' @param data new or old data for testing summary flags
+#' (data is from [get_source_extract_path(year, "CH")])
```
[the remark this answers is not in the document]
Comment (Jennit07) [reply R/produce_slf_ch_tests.R:6]: Yeah, i agree i think this is also from that line

File: R/produce_slf_ch_tests.R [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    # remove variables that won't be summed
+    dplyr::select(
+      .data$valid_chi,
+      .data$unique_chi,
+      .data$n_missing_chi,
+      .data$n_males,
+      .data$n_females,
+      .data$n_postcode,
+      .data$n_missing_postcode,
+      .data$missing_dob
```
Comment (Jennit07) [inline R/produce_slf_ch_tests.R:25]:
```suggestion
    dplyr::select(.data$valid_chi:.data$missing_dob)
```
I think this can be written like this following from other scripts

File: R/produce_slf_ch_tests.R [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    ) %>%
```
Comment (Jennit07) [inline R/produce_slf_ch_tests.R:26]: I've checked the CH tests script and i think there are some flags you are missing here. Have a look at B08X Care home tests I think you are missing tests for: CH provider, SC support and sc_send_lca which some of these can be put into a function

File: R/produce_slf_ch_tests.R [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#' Care Home All Episodes Tests
```
Comment (Jennit07) [inline R/produce_slf_ch_tests.R:1]:
```suggestion
#' Care Home year specific Episodes Tests
```
Which script are you doing tests for here? There are already tests for the all care home episode file see the function `produce_sc_ch_episodes_tests` which is used in the script `extract_all_sc_ch_tests` however, there are no tests for the Care home year specific scripts which your scripts can be applied to

File: R/produce_slf_ch_tests.R [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+#' @description Produce the test for the Care Home
+#'
+#' @param data new or old data for testing summary flags
+#' (data is from [get_source_extract_path(year, "CH")])
+#'
+#' @return a dataframe with a count of each flag.
+#'
+#' @export
+#' @family social care test functions
+produce_slf_ch_tests <- function(data) {
```
Comment (Jennit07) [inline R/produce_slf_ch_tests.R:12]:
```suggestion
produce_slf_ch_year_specific_tests <- function(data) {
```
Rename to something like this if this is the year specific care home tests?

**Review by Jennit07 — CHANGES_REQUESTED** [review]

**Comment by Moohan** [comment]
Need to add changes to `ch_adm_reason` to this PR, see #361 / #359

File: R/produce_slf_ch_tests.R (hunk 3 above)
Comment (lizihao-anu) [reply R/produce_slf_ch_tests.R]: Hi, @Jennit07. I found the test variables from SPSS script include those as follows.  Do I need to include all in the R script? Apologize if my memory is wrong. I remember James mentioned those variables starting with "sc_" are not needed. Can you please confirm that? Are there any variables I am missing? Thank you!

-     /n_ch_name_missing 
-     /n_ch_provider_1_to_5
-     /n_ch_provider_other
-     /n_ch_adm_reason_missing
-     /n_ch_nursing
-     /n_sc_living_alone_no
-     /n_sc_living_alone_yes 
-     /n_sc_living_alone_unknown
-     /n_sc_support_from_unpaid_carer_no
-     /n_sc_support_from_unpaid_carer_yes
-     /n_sc_support_from_unpaid_carer_unknown
-     /n_sc_social_worker_no 
-     /n_sc_social_worker_yes 
-     /n_sc_social_worker_unknown 
-     /n_sc_meals_no 
-     /n_sc_meals_yes 
-     /n_sc_meals_unknown 
-     /n_sc_day_care_no
-     /n_sc_day_care_yes
-     /n_sc_day_care_unknown

Comment (Jennit07) [reply R/produce_slf_ch_tests.R]: Hi @lizihao-anu yes, those variables listed are not needed. If you look at the SPSS script we want to test for beddays and costs you can use the function `calculate_measures` to help with this, see other tests such as acute, mental health and maternity to see how this is implemented. The script also checks how many episodes are in each health board where you can use the function `create_hb_test_flags`. Let me know if you need any help with this or if you would like me to talk through how the tests work as they can be complicated. Thanks

**Comment by lizihao-anu** [comment]
This new commit is related to #262, which is quite similar. So I decided to put the changes along with this pull.

**Comment by Jennit07** [comment]
> This new commit is related to #262, which is quite similar. So I decided to put the changes along with this pull.

Could you please split out the changes for home care into a new branch? It makes it easier to track PR's and ticking off what is finished on github

File: Production_scripts/Process_extracts/Produce_extract_tests/extract_home_care_tests.R [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+# Read current SLF episode file and filter for CH
+existing_data <- get_existing_data_for_tests(new_data = new_data)
+
+
+# Produce comparison-------------------------------------
+# Compare new file with existing slf data
+comparison <- produce_test_comparison(
+  produce_source_ch_tests(existing_data),
+  produce_source_ch_tests(new_data)
```
Comment (Jennit07) [inline Production_scripts/Process_extracts/Produce_extract_tests/extract_home_care_tests.R:30]: I think we can rename this function if this is also specific to home care tests. Maybe `produce_source_sc_tests`? If this is also specific to alarms telecare and sds data

**Review by Jennit07 — CHANGES_REQUESTED** [review]
A few comments

File: Production_scripts/Process_extracts/Produce_extract_tests/extract_home_care_tests.R [hunk 7]
```diff
@@ -0,0 +1,40 @@
+####################################################
+# Name of file - extract_home_care_tests.R
+# Original Authors - Zihao Li
+# Original Date - September 2022
+# Written/run on - RStudio Server
+# Version of R - 3.6.1
+# Description - Produce tests for source linkage files:
+#               home care processed file.
+#####################################################
+
```
[the remark this answers is not in the document]
Comment (lizihao-anu) [reply Production_scripts/Process_extracts/Produce_extract_tests/extract_home_care_tests.R:10]: Yeah. Sure. I will do it now

File: R/produce_slf_ch_tests.R [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+#' @description Produce the test for the Care Home
+#'
+#' @param data new or old data for testing summary flags
+#' (data is from [get_source_extract_path()])
+#'
+#' @return a dataframe with a count of each flag.
+#'
+#' @export
+#' @family social care test functions
+produce_slf_ch_year_specific_tests <- function(data) {
```
[the remark this answers is not in the document]
Comment (lizihao-anu) [reply R/produce_slf_ch_tests.R:12]: Yeah. Good spot!

File: Production_scripts/Process_extracts/Produce_extract_tests/extract_care_home_tests.R [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+comparison <- produce_test_comparison(
+  produce_source_sc_tests(existing_data),
+  produce_source_sc_tests(new_data)
+)
+
+
+# Produce Outfile----------------------------------------
+
+# Save test comparisons as an excel workbook
+write_tests_xlsx(comparison, "care_home_extract")
```
Comment (Jennit07) [inline Production_scripts/Process_extracts/Produce_extract_tests/extract_care_home_tests.R:37]:
```suggestion
write_tests_xlsx(comparison, "CH")
```
Use short hand recid here for saving output - done in another PR

File: R/produce_source_sc_tests.R [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+#' @param sum_mean_vars variables used when selecting 'all' measures from [calculate_measures()]
+#' @param max_min_vars variables used when selecting 'min-max' from [calculate_measures()]
+#'
+#' @return a dataframe with a count of each flag
+#' from [calculate_measures()]
+#' @export
+#'
+#' @family extract test functions
+#' @seealso calculate_measures
+produce_source_sc_tests <- function(data,
```
Comment (Jennit07) [inline R/produce_source_sc_tests.R:21]:
```suggestion
produce_source_ch_tests <- function(data,
```
We can split ch and hc out into a new function each as there are some different variables which are being measured. e.g hours for home care and `ch_provider`

File: R/produce_source_sc_tests.R [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+                                    sum_mean_vars = c("beddays", "cost", "yearstay"),
+                                    max_min_vars = c(
+                                      "record_keydate1", "record_keydate2",
+                                      "cost_total_net", "yearstay"
+                                    )) {
+  test_flags <- data %>%
+    # use functions to create HB and partnership flags
+    create_demog_test_flags() %>%
+    dplyr::mutate(n_episodes = 1) %>%
```
Comment (Jennit07) [inline R/produce_source_sc_tests.R:30]:
```suggestion
    dplyr::mutate(n_episodes = 1,
                  ch_name_missing = dplyr::if_else(is.na(.data$ch_name), 1, 0),
                  ch_provider_1_to_5 = dplyr::case_when(.data$ch_provider %in% c('1', '2', '3', '4', '5') ~ 1,
                                                        TRUE ~ 0),
                  ch_provider_other = dplyr::if_else(.data$ch_provider == '6', 1, 0),
                  ch_adm_reason_missing = dplyr::if_else(is.na(.data$ch_adm_reason), 1, 0)
                  ) %>%
```
Some missing variables can be added here. I think this code should work and produce the output

**Review by Jennit07 — CHANGES_REQUESTED** [review]
Looks good - a few comments

**Review by Jennit07 — APPROVED** [review]

**Comment by Jennit07** [comment]
Looks good to me! One thing to remember with the `new_data` is reading the zsav version until the rds is ready. Tested with the zsav version and this works fine. Happy to merge
